bazel/packaging/packaging.bzl

```diff
@@ -128,34 +131,130 @@ def _prepare_package_content(ctx, dynamic_loader_path = "/opt/redpanda/lib"):
         hwloc_distrib = hwloc_distrib,
         openssl = openssl,
         rpk_binary = ctx.file.rpk_binary,
-        shared_libraries = collections.uniq(shared_libraries),
+        shared_libraries = package_binaries.shared_libraries,
     )
 
+def _common_redpanda_package_cfg(ctx, package_content, fips_enabled, base_path = None):
+    def _path(path):
+        return "{}/{}".format(base_path, path) if base_path else path
+
+    binaries = []
+    if package_content.rp_util != None:
+        binaries.append({
+            "path": _path("bin"),
+            "name": "rp_util",
+            "source": package_content.rp_util.path,
+        })
+
+    if package_content.iotune != None:
+        binaries.append({
+            "path": _path("bin"),
+            "name": "iotune",
+            "source": package_content.iotune.path,
+        })
+
+    if package_content.hwloc_calc != None:
+        binaries.append({
+            "path": _path("bin"),
+            "name": "hwloc-calc",
+            "source": package_content.hwloc_calc.path,
+        })
+
+    if package_content.hwloc_distrib != None:
+        binaries.append({
+            "path": _path("bin"),
+            "name": "hwloc-distrib",
+            "source": package_content.hwloc_distrib.path,
+        })
+
+    if package_content.openssl != None:
+        binaries.append({
+            "path": _path("bin"),
+            "name": "openssl",
+            "source": package_content.openssl.path,
+        })
+
+    if package_content.rpk_binary != None:
+        binaries.append({
+            "path": _path("bin"),
+            "name": "rpk",
+            "source": package_content.rpk_binary.path,
+        })
+
+    files = []
+
+    for solib in package_content.shared_libraries:
+        files.append({
+            "path": _path("lib"),
+            "name": solib.basename,
+            "source": solib.path,
+        })
+
+    if fips_enabled:
+        files.append({
+            "path": _path("fips"),
+            "name": "fipsmodule.cnf",
+            "source": ctx.file.fips_config.path,
+        })
+        files.append({
+            "path": _path("fips"),
+            "name": "fips.so",
+            "source": ctx.file.fips_module.path,
+        })
+
+    return {
+        "binaries": binaries,
+        "package_files": files,
+        "package_dirs": [
+            _path("bin"),
+            _path("lib"),
+            _path("libexec"),
+            _path("config"),
+        ],
+        "owner": ctx.attr.owner,
+    }
+
+def _dir_package_configuration(ctx, package_content, fips_enabled):
+    cfg = _common_redpanda_package_cfg(ctx, package_content, fips_enabled)
+    cfg["binaries"].append({"path": "libexec", "name": "redpanda", "source": package_content.redpanda_binary.path})
```

Comment:
why does this need to be in libexec?

Reply:
the ducktape script creates a script in the `bin` dir that runs redpanda  from `libexec` using a loader this way the `bin/redpanda` stays as the redpanda executable path